e2e_local.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# These are the commands that will be sequentially run in the
+# container. The final one will be the test itself, which is supplied
+# as the first argument to this script
+#
+# Not every test requires `expect`, but it's not hurting anything to
+# install it for everything.
+#
+# Note that the `ci-studio-common.sh` file is baked into the image we
+# use.
+raw_commands=(". /opt/ci-studio-common/buildkite-agent-hooks/ci-studio-common.sh"
+              ".expeditor/scripts/setup_environment.sh DEV"
+              "hab pkg install --binlink --channel=stable core/expect"
+              "${*}")
+
+# Add a newline after every command, for feeding into the container.
+commands="${raw_commands[*]/%/\\n}"
```

Comment:
I think this should use `[@]` rather than `[*]`. Additionally, would it be simpler to append `;` to separate commands? I think that would eliminate the need for `${commands@E}` later. Eliminating that dependency would be nice, since it's a feature that doesn't exist in the version of `bash` that ships with macOS.

Reply:
Looks like `;` works fine, too. The `\n` approach was in keeping with what Buildkite actually executes, but it's functionally equivalent and (as you observe) a bit more cross-platform friendly.

I'll make the change.

Reply:
I looked at `*` vs `@` and it looks like it really does have to be `*`;

``` sh
 ${commands[@]/%/;}  # -> first 'command;' second 'command;' third 'command;'
 ${commands[*]/%/;}  # -> first 'command;' second 'command;' third 'command;'
"${commands[@]/%/;}" # -> 'first command;' 'second command;' 'third command;'
"${commands[*]/%/;}" # -> 'first command; second command; third command;'
```
Only the last one gives us what we want; a single string of delimited commands to pass into `bash -c`.

Reply:
If you only want to allow one command (with arguments) as input to `e2e_local.sh` and not have to put quotes around them; I agree. Using `@` is necessary if you want to allow more than one command on the CLI, but it would have to be invoked like `e2e_local.sh "cmd1 arg1" "cmd2 arg2"`

Reply:
I think for this iteration, I'll leave the code as-is, since I think it'll be a good forcing function to move us toward more standardization.

If that's trickier to do in practice, then we can easily modify things as laid out in these comments.